### PR TITLE
DDF-1181 moved attribute index filtering to UI as opposed to in the search endpoint

### PR DESCRIPTION
**Pull request by stustison**
@pklinef @andrew-fiedler

[Review on Reviewable](https://reviewable.io/reviews/codice/ddf-ui/78)

### CODE REVIEW

File: search-ui/search-endpoint/src/main/java/org/codice/ddf/ui/searchui/query/controller/SearchController.java [hunk 1]
```diff
@@ -377,7 +370,7 @@ private QueryResponse executeQuery(String sourceId, SearchRequest searchRequest,
     private QueryResponse getEmptyResponse(String sourceId) {
         // No query was specified
         QueryRequest queryRequest = new QueryRequestImpl(null, false,
-                Arrays.asList(sourceId), null);
+                Collections.singletonList(sourceId), null);
```
Comment (pklinef) [inline search-ui/search-endpoint/src/main/java/org/codice/ddf/ui/searchui/query/controller/SearchController.java:373]: Nice

File: search-ui/search-endpoint/src/main/java/org/codice/ddf/ui/searchui/query/controller/SearchController.java [hunk 2]
```diff
@@ -510,10 +503,11 @@ private JSONObject convertToJSON(MetacardType metacardType)
...
 
         for (AttributeDescriptor descriptor : metacardType.getAttributeDescriptors()) {
-            if (descriptor.isIndexed()) {
-                fields.put(descriptor.getName(),
-                        descriptor.getType().getAttributeFormat().toString());
-            }
+            JSONObject description = new JSONObject();
+            description.put("format", descriptor.getType().getAttributeFormat().toString());
+            description.put("indexed", descriptor.isIndexed());
+
```
Comment (pklinef) [inline search-ui/search-endpoint/src/main/java/org/codice/ddf/ui/searchui/query/controller/SearchController.java:509]: I would say we should add the other values in AttributeDescriptor but they are not currently used.

**Comment by pklinef** [comment]
https://tools.codice.org/wiki/display/DDF/Asynchronous+Queries#AsynchronousQueries-ResponseFormat needs to be updated.

@ricklarsen We still need to migrate the asynchronous pages to the UI documentation.

File: search-ui/standard/src/main/webapp/js/controllers/Filter.controller.js [hunk 3]
```diff
@@ -93,18 +93,19 @@ define([
                         _.each(pairs, function (pair) {
                             if (!_.contains(currentFieldNames, pair[0])) {
                                 // doesn't exist.  lets add.
+                                if (pair[1].indexed === true) {
```
Comment (andrewkfiedler) [inline search-ui/standard/src/main/webapp/js/controllers/Filter.controller.js]: Might as well just combine this with the previous if statement (with the index check first, as that's probably the cheaper operation).

**Comment by andrewkfiedler** [comment]
:+1: built and tested

**Comment by pklinef** [comment]
:+1:

**Comment by ricklarsen** [comment]
Moved Asynchronous pages to UI section of wiki.

**Pull request merged**